python/pyarrow/tests/test_compute.py

```diff
@@ -1051,6 +1052,31 @@ def test_pad():
     assert pc.utf8_rpad(arr, 3).tolist() == [None, 'á  ', 'abcd']
 
 
+def test_utf8_zfill():
+    assert pc.utf8_zfill is pc.utf8_zero_fill
+    arr = pa.array(['A', 'AB', 'ABC', None])
```

Comment:
Instead of writing the test examples by hand, perhaps validate that `pc.utf8_zero_fill` has the same semantics as `str.zfill`?